components/dashboard/src/components/MonacoEditor.tsx

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { useEffect, useRef } from "react";
+import * as monaco from "monaco-editor";
+
+export default function MonacoEditor(props: { classes: string, disabled?: boolean, language: string, value: string, onChange: (value: string) => void }) {
+  const containerRef = useRef<HTMLDivElement>(null);
+  const editorRef = useRef<monaco.editor.IStandaloneCodeEditor>();
+
+  useEffect(() => {
+    if (containerRef.current) {
```

Comment:
`if (!containerRef.current)` ?

Reply:
Hmm, I believe this is correct as is: You want the containing element (`containerRef.current`) to exist, so that you can mount a new editor (`editorRef.current`) on it.

Reply:
oops! I misread the refs and confused editorRef with it.